exploration-module/src/main/java/io/edpn/backend/exploration/adapter/persistence/entity/mapper/MybatisStationEntityMapper.java

```diff
@@ -0,0 +1,84 @@
+package io.edpn.backend.exploration.adapter.persistence.entity.mapper;
+
+import io.edpn.backend.exploration.adapter.persistence.entity.MybatisStationEntity;
+import io.edpn.backend.exploration.adapter.persistence.entity.MybatisSystemEntity;
+import io.edpn.backend.exploration.application.domain.LandingPadSize;
+import io.edpn.backend.exploration.application.domain.Station;
+import io.edpn.backend.exploration.application.dto.persistence.entity.StationEntity;
+import io.edpn.backend.exploration.application.dto.persistence.entity.mapper.StationEntityMapper;
+import io.edpn.backend.exploration.application.dto.persistence.entity.mapper.SystemEntityMapper;
+import lombok.RequiredArgsConstructor;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class MybatisStationEntityMapper implements StationEntityMapper<MybatisStationEntity> {
+
+    private final SystemEntityMapper<MybatisSystemEntity> systemEntityMapper;
+
+    private static Map<String, Integer> getLandingPads(Station station) {
+        Map<String, Integer> landingPads = new HashMap<>();
+        for (LandingPadSize landingPadSize : LandingPadSize.values()) {
+            if (landingPadSize.equals(LandingPadSize.UNKNOWN)) {
+                continue;
+            }
+            int amount = Optional.ofNullable(station.landingPads().get(landingPadSize)).orElse(0);
+            landingPads.put(landingPadSize.name(), amount);
+        }
+
+        return landingPads;
+    }
+
+    @Override
+    public Station map(StationEntity stationEntity) {
+        return new Station(
+                stationEntity.getId(),
+                stationEntity.getMarketId(),
+                stationEntity.getName(),
+                stationEntity.getType(),
+                stationEntity.getDistanceFromStar(),
+                Optional.ofNullable(stationEntity.getSystem()).map(systemEntityMapper::map).orElse(null),
+                Optional.ofNullable(stationEntity.getLandingPads())
+                        .map(Map::entrySet)
+                        .map(Set::stream)
+                        .map(stream -> stream
+                                .collect(Collectors.toMap(
+                                        k -> LandingPadSize.valueOf(k.getKey()),
+                                        Map.Entry::getValue,
+                                        (first, second) -> first
+                                )))
+                        .orElse(null),
```

Comment:
The whole stream incantation above makes me suspect it _could_ be simplified, but I'm not certain and don't have time to check the code out and try it our in an IDE. Also could the `orElse(null)` return an empty map instead?